From 9af7864fbd2fbaa244d350450ffced82bbb8d106 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Sun, 23 Jan 2022 20:09:57 +0100
Subject: [PATCH 445/446] arm64: dts: rockchip: rk356x move to lower pcie
 address space

It seems inbound address translation is broken to the address space
above 12G. Fix it by moving to the smaller address space below 4G for
now.

Signed-off-by: Peter Geis <pgwipeout@gmail.com>
---
 arch/arm64/boot/dts/rockchip/rk356x.dtsi | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk356x.dtsi b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
index cdb2b2e37..2856736b9 100644
--- a/arch/arm64/boot/dts/rockchip/rk356x.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
@@ -770,11 +770,18 @@ pcie2x1: pcie@fe260000 {
 		phys = <&combphy2_psq PHY_TYPE_PCIE>;
 		phy-names = "pcie-phy";
 		power-domains = <&power RK3568_PD_PIPE>;
+/*
 		reg = <0x3 0xc0000000 0x0 0x400000>,
 		      <0x0 0xfe260000 0x0 0x10000>,
 		      <0x3 0x3f800000 0x0 0x800000>;
 		ranges = <0x1000000 0x0 0x7f700000 0x3 0x3f700000 0x0 0x00100000
 			  0x2000000 0x0 0x40000000 0x3 0x00000000 0x0 0x3f700000>;
+*/
+		reg = <0x0 0xf6800000 0x0 0x00400000>,
+		      <0x0 0xfe260000 0x0 0x00010000>,
+		      <0x0 0xf5d00000 0x0 0x00100000>;
+		ranges = <0x02000000 0x0 0xf5000000 0x0 0xf4000000 0x0 0x01d00000>,
+			 <0x01000000 0x0 0xf5e00000 0x0 0xf5e00000 0x0 0x00100000>;
 		reg-names = "dbi", "apb", "config";
 		resets = <&cru SRST_PCIE20_POWERUP>;
 		reset-names = "pipe";
-- 
2.31.1

